hoomd/mpcd/ConcentricCylindersGeometry.h

```diff
@@ -0,0 +1,170 @@
+        }
+
+    //! Detect collision between the particle and the boundary
+    /*!
+     * \param pos Proposed particle position
+     * \param vel Proposed particle velocity
+     * \param dt Integration time remaining
+     *
+     * \returns True if a collision occurred, and false otherwise
+     *
+     * \post The particle position \a pos is moved to the point of reflection, the velocity \a vel
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
+        /*
+         * Check if particle is in bounds
+         */
+        const signed char sign = (char)((rsq < m_R0_sq) - (rsq > m_R1_sq));
+        // exit immediately if no collision is found or particle is not moving normal to the wall
+        // (since no new collision could have occurred if there is no normal motion)
+        if (sign == 0 || vel.x * vel.x + vel.y * vel.y == Scalar(0))
+            {
+            dt = Scalar(0);
+            return false;
+            }
+
+        /*
+         * Find the time remaining when the particle is collided with wall. This time is computed
+         * by backtracking the position, dot(pos, pos) = R^2.
+         */
+        const Scalar Rsq = (sign == 1) ? m_R0_sq : m_R1_sq;
+        const Scalar a = vel.x * vel.x + vel.y * vel.y;
+        const Scalar b = -Scalar(2) * (pos.x * vel.x + pos.y * vel.y);
+        const Scalar c = pos.x * pos.x + pos.y * pos.y - Rsq;
+        dt = (-b + sign * (slow::sqrt(b * b - Scalar(4) * a * c))) / (Scalar(2) * a);
+
+        // backtrack the particle for dt to get to point of contact
+        pos -= vel * dt;
+
+        // update velocity according to boundary conditions
+        // no-slip requires reflection of the tangential components.
+        if (m_no_slip)
+            {
+            vel = -vel;
+            if (sign == -1)
+                {
+                vel.x += Scalar(2) * m_w * -pos.y;
+                vel.y += Scalar(2) * m_w * pos.x;
+                }
+            }
+        // slip requires reflection of the normal components
+        else
+            {
+            vel.x -= Scalar(2) * pos.x * pos.x * vel.x / (Rsq);
+            vel.y -= Scalar(2) * pos.y * pos.y * vel.y / (Rsq);
+            }
+
+        return true;
+        }
+
+    //! Check if a particle is out of bounds
+    /*!
+     * \param pos Current particle position
+     * \returns True if particle is out of bounds, and false otherwise
+     */
+    HOSTDEVICE bool isOutside(const Scalar3& pos) const
+        {
+        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
+        return (rsq > m_R1_sq || rsq < m_R0_sq);
+        }
+
+    //! Get inner radius of the cylinder
+    /*!
+     * \returns Inner radius of the cylinder
+     */
+    HOSTDEVICE Scalar getR0() const
```

Comment:
Since this is new code, I try to name the getters to match how I will expose them to Python. This is a little more readable. It's totally fine to still use R0 internally in the class, since that is not public.
```suggestion
    HOSTDEVICE Scalar getInnerRadius() const
```